frameworks/keyed/sveltekit-5-runes-fine-grained/src/routes/+page.svelte

```diff
@@ -0,0 +1,196 @@
+<svelte:options runes={true} />
+
+<script>
+
+  let rowId = $state(1);
+  let data = $state([]);
+  let selected = $state(undefined);
+
+  const adjectives = [
+    "pretty",
+    "large",
+    "big",
+    "small",
+    "tall",
+    "short",
+    "long",
+    "handsome",
+    "plain",
+    "quaint",
+    "clean",
+    "elegant",
+    "easy",
+    "angry",
+    "crazy",
+    "helpful",
+    "mushy",
+    "odd",
+    "unsightly",
+    "adorable",
+    "important",
+    "inexpensive",
+    "cheap",
+    "expensive",
+    "fancy",
+  ];
+  const colours = [
+    "red",
+    "yellow",
+    "blue",
+    "green",
+    "pink",
+    "brown",
+    "purple",
+    "brown",
+    "white",
+    "black",
+    "orange",
+  ];
+  const nouns = [
+    "table",
+    "chair",
+    "house",
+    "bbq",
+    "desk",
+    "car",
+    "pony",
+    "cookie",
+    "sandwich",
+    "burger",
+    "pizza",
+    "mouse",
+    "keyboard",
+  ];
+
+  const add = () => (data = [...data, ...buildData(1000)]),
+    clear = () => {
+      data = [];
+    },
+    partialUpdate = () => {
+      const clone = data.slice();
+      for (let i = 0; i < data.length; i += 10) {
+          clone[i].label += " !!!";
+      }
+      
+      data = clone;
+    },
+    remove = (row) => {
+      const clone = data.slice();
+      clone.splice(clone.indexOf(row), 1);
+      data = clone;
+    },
+    run = () => {
+      data = buildData(1000);
+    },
+    runLots = () => {
+      data = buildData(10000);
+    },
+    swapRows = () => {
+      if (data.length > 998) {
+        const clone = data.slice();
+        const tmp = clone[1];
+        clone[1] = data[998];
+        clone[998] = tmp;
+        data = clone;
+      }
+    };
+
+  function _random(max) {
+    return Math.round(Math.random() * 1000) % max;
+  }
+
+  function createItem() {
+    let label = $state(`${adjectives[_random(adjectives.length)]} ${
+      colours[_random(colours.length)]
+    } ${nouns[_random(nouns.length)]}`);
+
+    return {
+      id: rowId++,
+      label,
```

Comment:
This isn't fine-grain. It needs to be inside a getter/setter. See https://github.com/krausest/js-framework-benchmark/pull/1494 for how you can use `$state` on classes instead.